homeassistant/components/firmata/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for Firmata binary sensor input."""
+
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.core import callback
+
+from .board import FirmataBoardPin
+from .const import CONF_NEGATE_STATE, DOMAIN, PIN_MODE_INPUT, PIN_MODE_PULLUP
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Firmata binary sensors."""
+    new_entities = []
+
+    boards = hass.data[DOMAIN]
+    board = boards[config_entry.entry_id]
+    for binary_sensor in board.binary_sensors:
+        binary_sensor_entity = FirmataDigitalBinaryInput(
+            hass, config_entry, **binary_sensor
+        )
+        new_binary_sensor = await binary_sensor_entity.setup_pin()
+        if new_binary_sensor:
+            new_entities.append(binary_sensor_entity)
+
+    async_add_entities(new_entities)
+
+
+class FirmataDigitalBinaryInput(FirmataBoardPin, BinarySensorEntity):
+    """Representation of a Firmata Digital Input Pin."""
+
+    async def setup_pin(self):
```

Comment:
Setting up the pin is a device api and not an entity api. I think this belongs in the board module.

Reply:
Yeah, I see what you mean, and that will make it easier to implement other multi-entity parts (like servos) later.

Though I am not sure what the best way to go about this is. Which parts should go in a new Entity class and which should go to the device class? Also, should the device class extend anything? Should the new `FirmataDigitalBinaryInput` class on the device side be called by the new entity class?

Reply:
Maybe put the pin API and other related device APIs in the board class? Each entity should have access to its board. That way the entity can interact with the board.

Reply:
I see. Will do.